client/my-sites/checkout/cart/cart-coupon.jsx

```diff
@@ -102,7 +102,11 @@ class CartCoupon extends React.Component {
 					onChange={ this.handleCouponInput }
 					value={ this.state.couponInputValue }
 				/>
-				<button type="submit" className="button">
+				<button
+					type="submit"
+					className="cart__button button"
+					disabled={ ! trim( this.state.couponInputValue ) }
```

Comment:
I prefer a more explicit check than coercing a string into a boolean - seems safer to use something like [lodash's `isEmpty`](https://lodash.com/docs/#isEmpty).